commit 4e190e72b6a7ba43e68bbf891e3b635065a19ebc
Author: Steve Glendinning <steve.glendinning@smsc.com>
Date:   Thu Feb 19 18:47:12 2015 +0000

    smsx95xx: fix crimes against truesize
    
    smsc95xx is adjusting truesize when it shouldn't, and following a recent patch from Eric this is now triggering warnings.
    
    This patch stops smsc95xx from changing truesize.
    
    Signed-off-by: Steve Glendinning <steve.glendinning@smsc.com>

diff --git a/drivers/net/usb/smsc95xx.c b/drivers/net/usb/smsc95xx.c
index f2d01cb6f958..95e48804ca13 100644
--- a/drivers/net/usb/smsc95xx.c
+++ b/drivers/net/usb/smsc95xx.c
@@ -82,6 +82,10 @@ static bool turbo_mode = true;
 module_param(turbo_mode, bool, 0644);
 MODULE_PARM_DESC(turbo_mode, "Enable multiple frames per Rx transaction");
 
+static bool truesize_mode = false;
+module_param(truesize_mode, bool, 0644);
+MODULE_PARM_DESC(truesize_mode, "Report larger truesize value");
+
 static int __must_check __smsc95xx_read_reg(struct usbnet *dev, u32 index,
 					    u32 *data, int in_pm)
 {
@@ -1971,7 +1975,8 @@ static int smsc95xx_rx_fixup(struct usbnet *dev, struct sk_buff *skb)
 				if (dev->net->features & NETIF_F_RXCSUM)
 					smsc95xx_rx_csum_offload(skb);
 				skb_trim(skb, skb->len - 4); /* remove fcs */
-				skb->truesize = size + sizeof(struct sk_buff);
+				if (truesize_mode)
+					skb->truesize = size + sizeof(struct sk_buff);
 
 				return 1;
 			}
@@ -1989,7 +1994,8 @@ static int smsc95xx_rx_fixup(struct usbnet *dev, struct sk_buff *skb)
 			if (dev->net->features & NETIF_F_RXCSUM)
 				smsc95xx_rx_csum_offload(ax_skb);
 			skb_trim(ax_skb, ax_skb->len - 4); /* remove fcs */
-			ax_skb->truesize = size + sizeof(struct sk_buff);
+			if (truesize_mode)
+				ax_skb->truesize = size + sizeof(struct sk_buff);
 
 			usbnet_skb_return(dev, ax_skb);
 		}
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1103_4e190e72b6a7ba43e68bbf891e3b635065a19ebc.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1103_4e190e72b6a7ba43e68bbf891e3b635065a19ebc.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
